browser/ui/views/frame/brave_browser_view.cc

```diff
@@ -896,6 +899,11 @@ void BraveBrowserView::AddedToWidget() {
     GetBrowserViewLayout()->set_vertical_tab_strip_host(
         vertical_tab_strip_host_view_.get());
   }
+
+  if (secondary_contents_web_view_) {
+    secondary_location_bar_ = SplitViewLocationBar::Create(
```

Comment:
this looks like a memory leak?

Reply:
I think the `SplitViewLocationBar` lives in a Widget, which is owned by the NativeWidget for the `secondary_contents_web_view_`, so it will be destroyed when the NativeWidget for `secondary_contents_web_view_` is destroyed

Reply:
Yes, @fallaciousreasoning is right. And the native widget of split view location bar will be destroyed when browser view's widget is destroyed.

Reply:
can you point me to exactly where this gets destroyed because it seems like we're creating another one and it's only going to delete one of them

Reply:
To my knowledge, there're two types of ownership model regarding `views::Widget`. By default we don't own the views::Widget, and `NativeWidget(Mac, Aura)` owns it. `NaitveWidget` is a abstraction of platform window handle like HWND or NSWIndow. As the mini location bar's parent is browser view, the native widget of location bar will be destroyed when native widget of browser view is destroyed by the OS window system.

(The other ownership model is a case where we explicitly mark we want own the Widget, but this is not the case of the mini url bar here)

```
NativeWidgetMac::~NativeWidgetMac() {
  if (ownership_ == Widget::InitParams::NATIVE_WIDGET_OWNS_WIDGET)
    delegate_.ClearAndDelete();
  else
    CloseNow();
}
```

Reply:
The way this is created is very confusing because you're passing in the parent, but it's not at all clear to the caller. The param isn't even named `parent` it's just `web_view`. Why aren't we just creating this similar to ToolbarView where it's much more obvious that the web view takes ownership?
```
auto location_bar = std::make_unique<LocationBarView>(
      browser_, browser_->profile(), browser_->command_controller(), this,
      display_mode_ != DisplayMode::NORMAL);
...
location_bar_ = container_view_->AddChildView(std::move(location_bar));
```

Reply:
That'd be because SplitViewLocationBar is not owned by the caller, BrowserView. SplitViewLocationBar is owned by the widget of SplitViewLocationBar. I believe this is a common way when we create a view that's owned by an additional widget, like bubble. For example, when BookmarkBubbleView hide its constructor and has the `static void ShowBubble`. 

But I agree that the parameter name is unclear. Let me change `web_view` => `parent_web_view`.

Reply:
`I believe this is a common way when we create a view that's owned by an additional widget` - can you point me to an example because this is definitely now how it is done in the examples I saw and as I said I think it's very confusing even if it's technically correct in terms of not leaking. It is not owned by the SplitViewLocationBar because that is also created in `SplitViewLocationBar::Create` and passed to the InitParams. It is owned by the parent https://github.com/brave/brave-core/pull/24884/files#diff-e0cfcdfcb7c79f006de509f703652eceba33a9ab07818ee2cb357493b92c3c0dR114 

There are various ways to do this including having a method that return the InitParams, having the method return the Widget , but the only other example I see like this is `VerticalTabStripWidgetDelegateView::Create` which has the same problem. In some cases the code to create the widget is similar, but it's not in a separate class. In some cases the code returns a unique_ptr to the widget or a unique_ptr to the view, but this particular implementation obscures ownership in a particularly confusing way imo and I can't find another example where the code to create the widget/view is in a separate class and it doesn't return any kind of ownership of the widget or view back to the caller.

Reply:
Moved widget and SplitViewLocationBar creation to BrowserView. Hopefully it reads better  now.